drivers/spi/spi_ll_stm32.c

```diff
@@ -723,6 +754,34 @@ static int wait_dma_rx_tx_done(const struct device *dev)
 	return res;
 }
 
+#ifdef CONFIG_SOC_SERIES_STM32H7X
+static bool buf_in_nocache(uintptr_t buf, size_t len_bytes)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(nocache_mem_regions); i++) {
+		const struct mem_region *mem_reg = &nocache_mem_regions[i];
+
+		const bool buf_within_bounds =
+			(buf >= mem_reg->start) && ((buf + len_bytes - 1) <= mem_reg->end);
```

Comment:
why you do not remove the `-1` from `mem_reg->end` and just:
```c
(buf + len_bytes) < mem_reg->end
```

Reply:
I did this on purpose. In general terms, this is checking if a number X is within a closed interval. If you would need to do this, you would naturally use <= and >=. I wanted to follow this pattern as it seems more intuitive. Otherwise, the bool variable is called "within_bounds" but it seems to not be doing exactly that. It seams more clear to me this way. In other contexts (most of them possibly), I would do as you say.